mlir/lib/Dialect/Transform/Transforms/TransformInterpreterPassBase.cpp

```diff
@@ -477,6 +477,14 @@ LogicalResult transform::detail::interpreterBaseInitializeImpl(
   if (parsedTransformModule && failed(mlir::verify(*parsedTransformModule)))
     return failure();
 
+  auto libraryRange = context->getOrLoadDialect<transform::TransformDialect>()
+                          ->getLibraryModules();
+  if (!transformLibraryFileName.empty() && !libraryRange.empty()) {
+    return emitError((*libraryRange.begin()).getLoc())
+           << "library already supplied through the dialect, cannot parse "
+              "another library as requested by pass flags";
```

Comment:
We could also allow both and merge the two libraries (as long as no symbols collide).